File: Graph/C++/11725 트리의 부모 찾기.cpp
#include <bits/stdc++.h>

using namespace std;

vector<vector<int>> graph(100001, vector<int>());
int answer[100001];

void dfs(int start) {
    for(int to : graph[start]) {
        if(answer[to] == 0) {
            answer[to] = start;
            dfs(to);
        }
    }
}

int main() {
    int n;
    cin >> n;
    for(int i = 0; i < n-1; i++) {
        int from; int to;
        cin >> from >> to;
        graph[from].push_back(to);
        graph[to].push_back(from);
    }
    dfs(1);
    for(int i = 2; i < n+1; i++) {
        cout << answer[i] << '\n';
    }
}

File: mvc1/_etc/C#/11725 트리의 부모 찾기.cs
using System;
using System.Collections.Generic;
using System.Text;
using static System.Console;


class Program
{
    static List<List<int>> graph = new List<List<int>>();
    static int[] answer = new int[100001];
    static StringBuilder sb = new StringBuilder();
    
    static void dfs(int start) {
        foreach(int to in graph[start]) {
            if(answer[to] == 0) {
                answer[to] = start;
                dfs(to);
            }
            
        }
        
        
    }
    
    static void Main() {
        string s = ReadLine();
        int n = int.Parse(s);
        for(int i = 0; i < n+1; i++) {
            graph.Add(new List<int>());
        }
        
        for(int i = 0 ; i < n-1; i++) {
            string[] fromto= ReadLine().Split(' '); 
            int from = int.Parse(fromto[0]), to = int.Parse(fromto[1]);
            graph[from].Add(to);
            graph[to].Add(from);
        }
        dfs(1);
        for(int i = 2 ; i < n+1; i++) {
            sb.Append(answer[i]);
            sb.Append('\n');
        }
        
        Write(sb.ToString());
        
    }
}
